### ISSUE TITLE
Indefinite resubmission of jobs.

### ISSUE THREAD

**Issue by tarekelgindy** [NREL/jade#37]
I ran across a problem today where jade continued to submit jobs even after the job had completed.
This resulted in an explosion of HPC submissions which may be a very large problem for users who encounter this bug.
This may have coincided with exceeding my HPC time allocation limit.

A snippet from one of the job .e files looks like this:
```
2021-06-19 07:28:54,995 - INFO [jade.jobs.cluster cluster.py:488] : Wrote config version 151 reason=promote_to_submitter hostname=r7i3n16
2021-06-19 07:28:55,056 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:78] : Created split config file jade_output_missing/config_batch_1.json with 20 jobs
2021-06-19 07:28:55,058 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/run_batch_1.sh
2021-06-19 07:28:55,061 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/medium_batch_1.sh
2021-06-19 07:28:55,062 - INFO [jade.hpc.hpc_manager hpc_manager.py:203] : Created submission script jade_output_missing/medium_batch_1.sh
2021-06-19 07:28:55,092 - INFO [jade.hpc.hpc_manager hpc_manager.py:212] : job 'medium_batch_1' with ID=7017599 submitted successfully
2021-06-19 07:28:55,093 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:458] : Assigned job_ID=7017599 name=medium_batch_1
2021-06-19 07:28:55,097 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:78] : Created split config file jade_output_missing/config_batch_2.json with 20 jobs
2021-06-19 07:28:55,098 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/run_batch_2.sh
2021-06-19 07:28:55,101 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/medium_batch_2.sh
2021-06-19 07:28:55,103 - INFO [jade.hpc.hpc_manager hpc_manager.py:203] : Created submission script jade_output_missing/medium_batch_2.sh
2021-06-19 07:28:55,131 - INFO [jade.hpc.hpc_manager hpc_manager.py:212] : job 'medium_batch_2' with ID=7017600 submitted successfully
2021-06-19 07:28:55,131 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:458] : Assigned job_ID=7017600 name=medium_batch_2
2021-06-19 07:28:55,135 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:78] : Created split config file jade_output_missing/config_batch_3.json with 20 jobs
2021-06-19 07:28:55,136 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/run_batch_3.sh
2021-06-19 07:28:55,139 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/medium_batch_3.sh

...

2021-06-19 09:59:08,595 - INFO [jade.hpc.hpc_manager hpc_manager.py:212] : job 'large_batch_81' with ID=7023774 submitted successfully
2021-06-19 09:59:08,595 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:458] : Assigned job_ID=7023774 name=large_batch_81
2021-06-19 09:59:08,597 - INFO [jade.hpc.hpc_submitter hpc_submitter.py:78] : Created split config file jade_output_missing/config_batch_82.json with 14 jobs
2021-06-19 09:59:08,598 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/run_batch_82.sh
2021-06-19 09:59:08,600 - INFO [jade.utils.utils utils.py:64] : Writing jade_output_missing/bigmem_batch_82.sh
2021-06-19 09:59:08,600 - INFO [jade.hpc.hpc_manager hpc_manager.py:203] : Created submission script jade_output_missing/bigmem_batch_82.sh
2021-06-19 09:59:08,611 - ERROR [jade.hpc.hpc_manager hpc_manager.py:216] : Failed to submit job 'bigmem_batch_82': result=Status.ERROR: sbatch: error: Batch job submission failed: Invalid qos specification

2021-06-19 09:59:08,615 - INFO [jade.jobs.cluster cluster.py:488] : Wrote config version 302 reason=demote_from_submitter hostname=r7i3n16
Traceback (most recent call last):
  File "/home/telgindy/miniconda3/bin/jade", line 10, in <module>
    sys.exit(cli())
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/cli/try_submit_jobs.py", line 55, in try_submit_jobs
    status = mgr.submit_jobs(cluster)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/jobs/job_submitter.py", line 155, in submit_jobs
    is_complete = self._submit_to_hpc(cluster)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/jobs/job_submitter.py", line 385, in _submit_to_hpc
    if hpc_submitter.run():
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/utils/timing_utils.py", line 26, in timed_
    return _timed(func, logger.debug, *args, **kwargs)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/utils/timing_utils.py", line 33, in _timed
    result = func(*args, **kwargs)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/hpc/hpc_submitter.py", line 125, in run
    self._submit_batches(queue, group, blocked_jobs, submitted_jobs)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/hpc/hpc_submitter.py", line 180, in _submit_batches
    self._submit_batch(queue, submission_group, batch)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/hpc/hpc_submitter.py", line 200, in _submit_batch
    queue.submit(async_submitter)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/jobs/job_queue.py", line 231, in submit
    self._run_job(job)
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/jobs/job_queue.py", line 121, in _run_job
    job.run()
  File "/home/telgindy/miniconda3/lib/python3.7/site-packages/jade/hpc/hpc_submitter.py", line 447, in run
    raise ExecutionError("Failed to submit name={self._name}")
jade.exceptions.ExecutionError: Failed to submit name={self._name}
2021-06-19 09:59:08,711 - ERROR [jade.cli.run_jobs run_jobs.py:89] : Failed to run 'jade try-submit-jobs jade_output_missing' ret=1

```

Happy to provide more error outputs if it helps

**Comment by daniel-thom** [comment]
This is a JADE bug. It was triggered by the fact that `qos` value was set to `high` for one submission group and the allocation had consumed all of its allotted hours. This caused the SLURM submission to fail. Then JADE raised a generic exception without updating the jobs that had been submitted. This caused future submitters to resubmit the same jobs over and over again.

This would have only happened in configurations with multiple submission groups where one invalid submission group follows  valid submission groups. It could not happen in configurations with one submission group (default).

Fixed in v0.4.3.

**Comment by daniel-thom** [comment]
A variation of this could have occurred with only one submission group. If there were network problems in the HPC where one or more sbatch commands in a batch and then one failed, the successfully submitted jobs would have been resubmitted. Unlike this case, the jobs would likely only have been resubmitted one extra time assume the transient errors did not occur the next time. This is still obviously very problematic.

The users would hopefully have detected this when it happened. If it occurred when the user ran `jade submit-jobs` then it would have dumped a big stacktrace to the console. If it occurred in a deferred submission that part wouldn't have been as obvious. However, `jade show-results` and `jade show-status` would have displayed errors.

I have a script that can detect whether this error occurred on a past job submission and can make it available.

Version v0.4.4 adds handling of transient errors.
